### ISSUE TITLE
Lokinet UI

### ISSUE THREAD

**Issue by KeeJef** [loki-project/loki-network#849]
We should establish an unified Lokinet UI for Windows and Mac (and eventually Linux) operating systems, i suggest we use inspiration from Private internet access UI which is pretty stock standard in the VPN business. have posted various screenshots below. 

Unconnected

![PIA1](https://user-images.githubusercontent.com/27277414/65878117-a015d980-e3d0-11e9-8949-9e9305f549a9.png)

Connected

![PIA2revised](https://user-images.githubusercontent.com/27277414/65878118-a0ae7000-e3d0-11e9-8ed5-891b507890ca.png)

Settings Hamburger

![PIA3revised](https://user-images.githubusercontent.com/27277414/65878119-a0ae7000-e3d0-11e9-9ede-aad10410cf92.png)

Large settings 

![PIA4revised](https://user-images.githubusercontent.com/27277414/65878120-a0ae7000-e3d0-11e9-923b-91a1d6248efa.png)

**Comment by majestrate** [comment]
Yea we do need to start on the qt frontend ui, the real question is how much code can we reuse across all the platforms ( i am guessing we would be able to target everything with qt). A responsive QML based application would suffice at the very minimum for both desktop and mobile targets. ideally it should be one single application. qt has dbus stuff so talking to network manager would be easier for linux, windows would be easier too so we won't need a win32 specific code base for the UI, osx desktop would also benefit. making a frontend for each platform is a no go, it would make too much work and fragment the ui code.

**Comment by despair86** [comment]
this is doable, but it will probably catch flak with the macOS user base for not using native widgets

I'll probably spin off the mono winforms codebase into its own project and maintain it separately once the new ui lands

**Comment by despair86** [comment]
furthermore, once the new qml ui lands, I can script the installer to provide the qt5 ui in-box, and grab the old mono ui at runtime by user decision

**Comment by despair86** [comment]
about freakin time

that's like the #1 or #2 complaint of qt apps on powermac

**Comment by despair86** [comment]
to be fair, they say this of electron too, but they put up with that because it's so much easier to write ui in js

**Comment by despair86** [comment]
nice to have but completely optional: qt4 source compatibility (somewhat like the way mumble does)

**Comment by michael-loki** [comment]
@majestrate i was actually hoping that network extensions will end up being supported on macOS too, and then the iOS app would be native on both that way round. 

I’m not sure if qt is necessarily the right choice or not.

**Comment by jagerman** [comment]
For reference

PIA on mac: https://cdn.discordapp.com/attachments/465843308187484160/628388749134528525/Screen_Shot_2019-10-01_at_10.32.21.png

and linux: https://cdn.discordapp.com/attachments/465843308187484160/628405905771331584/Screenshot_from_2019-10-01_11-39-04-cropped.png


Some stuff to include here:

- Huge on/off button (at least 2x the side of PIA's :wink:  )
- your local .loki address
- Traffic stats
- some network stats like # of known routers

Are there other pieces that might be useful?

**Comment by neuroscr** [comment]
Definitely need Qt for Linux, so we'll use that for all Desktop platforms to keep management over time (maintenance) easy

**Comment by neuroscr** [comment]
Aiming for Jan 6
